ts/main.ts

```diff
@@ -28,9 +37,93 @@ document.addEventListener('DOMContentLoaded', () => {
     data.entries.unshift(newEntry);
     data.nextEntryId++;
 
+    const $entry = renderEntry(newEntry);
+    $entriesList.prepend($entry);
+
     $photoPreview.setAttribute('src', 'images/placeholder-image-square.jpg');
     $form.reset();
 
     savedData();
+    toggleNoEntries();
+    viewSwap('entries');
+  });
+
+  data.entries.forEach((entry) => {
+    const $entry = renderEntry(entry);
+    $entriesList.appendChild($entry);
   });
+
+  toggleNoEntries();
+  viewSwap(data.view);
+
+  $newEntryButton.addEventListener('click', (event) => {
+    event.preventDefault();
+    viewSwap('entry-form');
+  });
+
+  if ($entriesLink) {
+    $entriesLink.addEventListener('click', (event) => {
+      event.preventDefault();
+      viewSwap('entries');
+    });
+  }
 });
+
+function renderEntry(entry: JournalEntry): HTMLElement {
+  const $entryItem = document.createElement('li');
+  $entryItem.className = 'entry-item';
+
+  const $entryImage = document.createElement('div');
+  $entryImage.className = 'entry-image';
+  const $img = document.createElement('img');
+  $img.setAttribute('src', entry.photoUrl);
+  $img.setAttribute('alt', 'Entry Image');
+  $entryImage.appendChild($img);
+
+  const $entryContent = document.createElement('div');
+  $entryContent.className = 'entry-content';
+  const $title = document.createElement('h3');
+  $title.textContent = entry.title;
+  const $notes = document.createElement('p');
+  $notes.textContent = entry.notes;
+  $entryContent.appendChild($title);
+  $entryContent.appendChild($notes);
+
+  $entryItem.appendChild($entryImage);
+  $entryItem.appendChild($entryContent);
+
+  return $entryItem;
+}
+
+function toggleNoEntries(): void {
+  const $entriesList = document.getElementById(
+    'entries-list',
+  ) as HTMLUListElement;
+  const $noEntries = document.getElementById('no-entries') as HTMLElement;
```

Comment:
Why not query for these outside of the function? If these are elements in your HTML that are always present, we don't need to query for them in a function because we know they will be there. If we have it in the function we are adding more work to have to keep requerying the DOM for something that is permanent.

tldr; if the elements are permanently present on the HTML, move these to the top of the file